cmd/storage-rest-client.go

```diff
@@ -148,6 +148,10 @@ func (client *storageRESTClient) IsOnline() bool {
 	return atomic.LoadInt32(&client.connected) == 1
 }
 
+func (client *storageRESTClient) Hostname() string {
+	return client.endpoint.Host
```

Comment:
This is perhaps not needed you can just use . String() to get the full URL avoids adding new methods are POSIX interface.

Reply:
And extract the Hostname() ?

Reply:
It seems a bit fragile to do it that way IMO.

Reply:
Okay @klauspost fair enough I don't see a big deal in avoiding it.